core/src/main/java/io/grpc/internal/InternalSubchannel.java

```diff
@@ -221,21 +205,24 @@ private Runnable startNewTransport() {
     }
     pendingTransport = transport;
     transports.add(transport);
-    return transport.start(new TransportListener(transport, address));
+    Runnable runnable = transport.start(new TransportListener(transport, address));
+    if (runnable != null) {
+      channelExecutor.executeLater(runnable);
```

Comment:
I'm not sure if this is an improvement. The transport will probably always need to return a `Runnable`. And the runnable doesn't need to be synchronized with anything else. Basically, I don't feel like the abstractions provided by the `ChannelExecutor` provide any value in this case and instead increases chance of bug.